src/Promotion/Decorator/PromotionSubjectCouponEligibilityCheckerDecorator.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Advanced Promotion plugin for Sylius.
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusAdvancedPromotionPlugin\Promotion\Decorator;
+
+use MonsieurBiz\SyliusAdvancedPromotionPlugin\Entity\PromotionCouponsAwareInterface;
+use Sylius\Component\Promotion\Checker\Eligibility\PromotionCouponEligibilityCheckerInterface;
+use Sylius\Component\Promotion\Checker\Eligibility\PromotionEligibilityCheckerInterface;
+use Sylius\Component\Promotion\Checker\Eligibility\PromotionSubjectCouponEligibilityChecker;
+use Sylius\Component\Promotion\Model\PromotionInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Symfony\Component\DependencyInjection\Attribute\AsDecorator;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\DependencyInjection\Attribute\AutowireDecorated;
+
+/**
+ * @SuppressWarnings(PHPMD.LongClassName)
+ */
+#[AsDecorator('sylius.promotion_subject_coupon_eligibility_checker')]
+final class PromotionSubjectCouponEligibilityCheckerDecorator implements PromotionEligibilityCheckerInterface
+{
+    public function __construct(
+        #[AutowireDecorated]
+        private readonly PromotionSubjectCouponEligibilityChecker $promotionSubjectCouponEligibilityChecker,
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏same comment, use the interface here

```suggestion
        private readonly PromotionEligibilityCheckerInterface $promotionSubjectCouponEligibilityChecker,
```